src/Raven.Pal/src/posix/maconly.c

```diff
@@ -68,4 +69,19 @@ rvn_test_storage_durability(
     return SUCCESS; /* windows and mac are always true */
 }
 
+PRIVATE int32_t
+rvn_pipe_for_child(int descriptors[2])
+{
+    int32_t result = FAIL;
+    while ((result = pipe(descriptors)) < 0 && errno == EINTR);
+    if (result != 0)
+        return FAIL_CREATE_PIPE;
+    
+    while ((result = fcntl(descriptors[0], F_SETFD, FD_CLOEXEC)) < 0 && errno == EINTR);
+    if (result == 0)
+        while ((result = fcntl(descriptors[1], F_SETFD, FD_CLOEXEC)) < 0 && errno == EINTR);
+
```

Comment:
Shouldn't we be closing the `descriptors` if there is a failure here?

Reply:
https://github.com/ravendb/ravendb/pull/9723/files/3406eb2840bcaf009737f0a829f0da841e86e133#diff-26816b18cf670fbae58f7f07ce0927fdR258